### PR TITLE
chore: merge with develop branch and bug fix

### PR DESCRIPTION
**Pull request by pythonberg1997**

### CODE REVIEW

File: contracts/BC_fusion/TokenRecoverPortal.sol [hunk 1]
```diff
@@ -93,6 +95,10 @@ contract TokenRecoverPortal is ReentrancyGuardUpgradeable, System {
     // This event is triggered whenever a call to #recover succeeds.
     event TokenRecoverRequested(bytes32 tokenSymbol, address account, uint256 amount);
 
+    function initialize() external initializer onlyCoinbase onlyZeroGasPrice {
+        approvalAddress = INIT_APPROVAL_ADDRESS;
```
Comment (zjubfd) [inline contracts/BC_fusion/TokenRecoverPortal.sol:99]: This `address public assetProtector = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa` need to change too.

File: contracts/BC_fusion/TokenRecoverPortal.sol [hunk 2]
```diff
@@ -17,17 +17,19 @@ import "./lib/Utils.sol";
...
     using Utils for string;
     using Utils for bytes;
 
-    /*----------------- init parameters -----------------*/
+    /*----------------- constants -----------------*/
     string public constant SOURCE_CHAIN_ID = "Binance-Chain-Ganges";
-    address public approvalAddress = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa;
-    bytes32 public merkleRoot = 0x0000000000000000000000000000000000000000000000000000000000000000;
-    bool public merkleRootAlreadyInit = false;
+    address public constant INIT_APPROVAL_ADDRESS = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa;
```
Comment (zjubfd) [inline contracts/BC_fusion/TokenRecoverPortal.sol:26]: no need to define a const here as it is only used for once.

File: contracts/BC_fusion/StakeCredit.sol [hunk 3]
```diff
@@ -159,6 +181,10 @@ contract StakeCredit is System, Initializable, ReentrancyGuardUpgradeable, ERC20
         uint256 bnbAmount = msg.value;
         uint256 _commission = (bnbAmount * uint256(commissionRate)) / COMMISSION_RATE_BASE;
         uint256 _reward = bnbAmount - _commission;
+
+        uint256 dayIndex = block.timestamp / 1 days;
+        totalPooledBNBRecord[dayIndex] = totalPooledBNB;
```
Comment (zjubfd) [inline contracts/BC_fusion/StakeCredit.sol:186]: move down this logic after `totalPooledBNB += _reward`

File: contracts/BC_fusion/StakeCredit.sol [hunk 4]
```diff
@@ -203,24 +229,51 @@ contract StakeCredit is System, Initializable, ReentrancyGuardUpgradeable, ERC20
...
+        return count;
+    }
+
+    /**
+     * @return the sum of first `number` requests' BNB locked in delegator's unbond queue.
+     */
+    function lockedBNBs(address delegator, uint256 number) public view returns (uint256) {
+        // number == 0 means all
+        // number should not exceed the length of the queue
+        if (_unbondRequestsQueue[delegator].length() == 0) revert NoUnbondRequest();
```
Comment (zjubfd) [inline contracts/BC_fusion/StakeCredit.sol:270]: return 0 directly

File: contracts/BC_fusion/StakeCredit.sol [hunk 5]
```diff
@@ -203,24 +229,51 @@ contract StakeCredit is System, Initializable, ReentrancyGuardUpgradeable, ERC20
...
+     * @return the total number of delegator's claimable unbond requests.
      */
-    function lockedBNBs(address delegator) public view returns (uint256) {
+    function claimableUnbondRequest(address delegator) public view returns (uint256) {
         uint256 length = _unbondRequestsQueue[delegator].length();
-        if (length == 0) {
-            return 0;
+        uint256 count;
+        for (uint256 i; i < length; ++i) {
+            bytes32 hash = _unbondRequestsQueue[delegator].front();
```
Comment (zjubfd) [inline contracts/BC_fusion/StakeCredit.sol:253]: you will always get the front item, as you never pop it.

File: contracts/BC_fusion/StakeCredit.sol (hunk 4 above)
Comment (Pythonberg1997) [reply contracts/BC_fusion/StakeCredit.sol:270]: Done

File: contracts/BC_fusion/StakeCredit.sol (hunk 5 above)
Comment (Pythonberg1997) [reply contracts/BC_fusion/StakeCredit.sol:253]: Fixed

**Pull request merged**